pkg/planner/core/plan_cache.go

```diff
@@ -163,8 +163,8 @@ func GetPlanFromSessionPlanCache(ctx context.Context, sctx sessionctx.Context,
 		cacheEnabled = sctx.GetSessionVars().EnablePreparedPlanCache
 	}
 	stmtCtx.UseCache = stmt.StmtCacheable && cacheEnabled
-	if !stmt.StmtCacheable && stmt.UncacheableReason != "" {
-		stmtCtx.SetSkipPlanCache(errors.New(stmt.UncacheableReason))
+	if stmt.UncacheableReason != "" {
```

Comment:
why remove `!stmt.StmtCacheable`?

Reply:
Because it has been checked outside.

Reply:
We need to print the warning even if it is uncacheable.